src/mozilla_taskgraph/worker_types.py

```diff
@@ -49,7 +52,7 @@
                 Optional(
                     "environments",
                     description="Environment variables to pass into the build.",
-                ): dict,
+                ): [{Required("mapped_to"): str, Optional("value"): taskref_or_string}],
```

Comment:
It might be worth adding a comment that points at whatever Bitrise docs exist for this (or maybe for `build_params` overall).